specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-06-01-preview/ConfidentialWatchlists.json

```diff
@@ -0,0 +1,441 @@
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/DeleteConfidentialWatchlist.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Delete a confidential watchlist.",
+        "operationId": "ConfidentialWatchlists_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/WatchlistAlias"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Azure-AsyncOperation": {
+                "description": "Contains the status URL on which clients are expected to poll the status of the delete operation.",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates or updates a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlist.json"
+          },
+          "Creates or updates a confidential watchlist and bulk creates watchlist items.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlistAndWatchlistItems.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Create or update a confidential Watchlist and its Watchlist Items (bulk creation, e.g. through text/csv content type). To create a Watchlist and its Items, we should call this endpoint with either rawContent or a valid SAR URI and contentType properties. The rawContent is mainly used for small watchlist (content size below 3.8 MB). The SAS URI enables the creation of large watchlist, where the content size can go up to 500 MB. The status of processing such large file can be polled through the URL returned in Azure-AsyncOperation header.",
```

Comment:
>its Watchlist Items (bulk creation, e.g. through text/csv content type). [](http://example.com/codeflow?start=70&length=72)

Does this PUT api create Microsoft.SecurityInsights/confidentialWatchlists/WatchlistItems, which is a child of confidentialWatchlists? Creating child resources implicitly is not ideal. e.g. it can bypass RBAC, no Azure policy support, etc.

Reply:
Yes it does. Discussed with ARM back in 2020 if there is any alternative to what we do. It seems not, and the _deep put_ was not supported yet (https://armwiki.azurewebsites.net/api_contracts/NestedProvisioning.html?q=Deep%20PUT).

Reply:
Yes deep put has been deprecated. ARM team is working on an alternative, but it is very early stage AFAIK.